main/src/cgeo/geocaching/CacheDetailActivity.java

```diff
@@ -1553,13 +1553,14 @@ private void uploadPersonalNote() {
                 public void call() {
                     final IConnector con = ConnectorFactory.getConnector(cache);
                     if (con.supportsPersonalNote()) {
-                        con.uploadPersonalNote(cache);
+                        final boolean success = con.uploadPersonalNote(cache);
+
+                        final Message msg = Message.obtain();
+                        final Bundle bundle = new Bundle();
+                        bundle.putString(SimpleCancellableHandler.MESSAGE_TEXT, CgeoApplication.getInstance().getString(success ? R.string.cache_personal_note_upload_done : R.string.cache_personal_note_upload_error));
+                        msg.setData(bundle);
+                        myHandler.sendMessage(msg);
```

Comment:
In CacheDetailActivity the upload button is only shown if the connector supports it.
See line 1474